index.js

```diff
@@ -82,87 +74,51 @@ function createDecoder(decoderLayers) {
-    })
-  );
-  autoencoder.add(
-    tf.layers.dense({
-      units: 64,
-      activation: "relu",
-    })
-  );
-  autoencoder.add(
-    tf.layers.dense({
-      units: 16,
-      activation: "relu",
-    })
-  );
-  autoencoder.add(
-    tf.layers.dense({
-      units: 4,
-      activation: "sigmoid",
-    })
-  );
-  // How do I start from here?
-  // Decoder
-
-  let decoderLayers = [];
-  decoderLayers.push(
-    tf.layers.dense({
-      units: 16,
-      activation: "relu",
-    })
-  );
-  decoderLayers.push(
-    tf.layers.dense({
-      units: 64,
-      activation: "relu",
-    })
-  );
-  decoderLayers.push(
-    tf.layers.dense({
-      units: 128,
-      activation: "relu",
-    })
-  );
-  decoderLayers.push(
-    tf.layers.dense({
-      units: 256,
-      activation: "relu",
-    })
-  );
-  decoderLayers.push(
-    tf.layers.dense({
-      units: W * W,
-      activation: "sigmoid",
-    })
-  );
-
-  for (let layer of decoderLayers) {
-    autoencoder.add(layer);
-  }
 
-  // const learningRate = 0.001;
-  // const optimizer = tf.train.adam(learningRate);
+// encoder
+  const encoder_input = tf.input({shape: [W*W], name: "encoder_input"});
+ // const l0 = tf.layers.flatten().apply(encoder_input);
+  const l1 = tf.layers.dense({units: 128, activation: "relu"}).apply(encoder_input);
+  const l2 = tf.layers.dense({units: 64, activation: "relu"}).apply(l1);
+  const l3 = tf.layers.dense({units: 16, activation: "relu"}).apply(l2);
+  const l4 = tf.layers.dense({units: 4, activation: "relu"}).apply(l3);
+  let encoded = tf.layers.dense({units: 2, activation: "relu", name: "encoder_output"}).apply(l4);
+  
+  let encoder = tf.model({inputs: encoder_input, outputs: encoded, name: "encoder"});
+  console.log(`Encoder Summary: ${encoder.summary()}`);
+
+  const decoder_input = tf.input({shape: [2]});
+  let decoder = tf.layers.dense({units: 4, activation: "relu", name: "decoder_input"}).apply(decoder_input);
+  const l6 = tf.layers.dense({units: 16, activation: "relu"}).apply(decoder);
+  const l7 = tf.layers.dense({units: 64, activation: "relu"}).apply(l6);
+  const l8 = tf.layers.dense({units: 128, activation: "relu"}).apply(l7);
+  let decoded = tf.layers.dense({units: 784, activation: "sigmoid", name: "decoder_output"}).apply(l8);
+
+  decoder = tf.model({inputs: decoder_input, outputs: decoded, name: "decoder"});
+  console.log(`Decoder Summary: ${decoder.summary()}`);
+
+
+  const auto = tf.input({shape: [W*W]});
+  encoded = encoder.apply(auto);
+  decoded = decoder.apply(encoded);
+
+  const autoencoder = tf.model({inputs: auto, outputs: decoded, name: "autoencoder"});
+  console.log(`Autoencoder Summary: ${autoencoder.summary()}`);
+
+ 
+  const learningRate = 0.0001;
+  const optimizer = tf.train.adam(learningRate, 0.000001); // adam(learning_rate, decay)
+  
   autoencoder.compile({
     optimizer: "adam",
     loss: "meanSquaredError",
   });
-  return { decoderLayers, autoencoder };
+  decoder.compile({
```

Comment:
I was wondering when I built this if the decoder has to be "compiled" with an optimizer and loss given I'm not training it, only using it for inference?